### PR TITLE
Updates to readme and changelog

### PR DESCRIPTION
**Pull request by LisaDelaney**

### CODE REVIEW

**Comment by saadrahim** [comment]
@cgmb @jzuniga-amd @tfalders @qjojo Can this PR be reviewed?

**Comment by saadrahim** [comment]
@jzuniga-amd can your team please review and merge this PR?

**Comment by jzuniga-amd** [comment]
Same comments from the rocSOLVER PR apply here. We have not seen any changes there.

**Comment by LisaDelaney** [comment]
> Same comments from the rocSOLVER PR apply here. We have not seen any changes there.

@jzuniga-amd how does this look? (We're paring down the number of categories in the changelog going forward.)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,241 +1,328 @@
-# Change Log for hipSOLVER
+# Changelog for hipSOLVER
 
+Documentation for hipSOLVER is available at
+[https://rocm.docs.amd.com/projects/hipSOLVER/en/latest/](https://rocm.docs.amd.com/projects/hipSOLVER/en/latest/).
 Full documentation for hipSOLVER is available at the [hipSOLVER Documentation](https://rocm.docs.amd.com/projects/hipSOLVER/en/latest/index.html).
```
Comment (tfalders) [inline CHANGELOG.md]: Shouldn't this line be removed? It's redundant with the previous line.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,241 +1,328 @@
...
+* Compatibility-only functions:
+  * `csrlsvchol`:
+    * `hipsolverSpScsrlsvcholHost`, `hipsolverSpDcsrlsvcholHost`
+    * `hipsolverSpScsrlsvchol`, `hipsolverSpDcsrlsvchol`
+* rocSPARSE and SuiteSparse are now optional dependencies for hipSOLVER (rocSOLVER backend
+  only)
+  * Use the `BUILD_WITH_SPARSE` CMake option to enable functionality for the `hipsolverSp` API (on by
+    default)
+* hipSPARSE is now an optional dependency for `hipsolver-test`. Use the `BUILD_WITH_SPARSE` CMake
+  option to enable hipsolverSp API tests (on by default)
```
Comment (tfalders) [inline CHANGELOG.md]: For consistency with similar changes, it seems like the part about `BUILD_WITH_SPARSE` should be on a separate line.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,241 +1,328 @@
...
+* Added hipBLAS as an optional dependency to `hipsolver-test`
+  * You can use the `BUILD_HIPBLAS_TESTS` CMake option to test the compatibility between hipSOLVER
+    and hipBLAS
+
+### Changes
+
+* The `hipsolverOperation_t` type is now an alias of `hipblasOperation_t`
+* The `hipsolverFillMode_t` type is now an alias of `hipblasFillMode_t`
+* The `hipsolverSideMode_t` type is now an alias of `hipblasSideMode_t`
+* Relax array length requirements for GESVDA
```
Comment (tfalders) [inline CHANGELOG.md]: This line doesn't appear to be present in the original changelog for 6.0. It should be removed.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,241 +1,328 @@
...
 
-### Fixed
-- Fixed tests for hipsolver info updates in ORGBR/UNGBR, ORGQR/UNGQR,
-  ORGTR/UNGTR, ORMQR/UNMQR, and ORMTR/UNMTR.
+### Fixes
 
+* Tests for hipSOLVER info updates in `ORGBR/UNGBR`, `ORGQR/UNGQR`, `ORGTR/UNGTR`,
+  `ORMQR/UNMQR`, and `ORMTR/UNMTR`
+* Incorrect singular vectors returned from GESVDA
```
Comment (tfalders) [inline CHANGELOG.md]: This line doesn't appear to be present in the original changelog for 6.0. It should be removed.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,241 +1,328 @@
...
+    * `hipsolverSpotrsBatched`, `hipsolverDpotrsBatched`, `hipsolverCpotrsBatched`, and
+      `hipsolverZpotrsBatched`
+    * `hipsolverSpotrsBatched_bufferSize`, `hipsolverDpotrsBatched_bufferSize`,
+      `hipsolverCpotrsBatched_bufferSize`, and `hipsolverZpotrsBatched_bufferSize`
+  * `potri`
+    * `hipsolverSpotri`, `hipsolverDpotri`, `hipsolverCpotri`, and `hipsolverZpotri`
+    * `hipsolverSpotri_bufferSize`, `hipsolverDpotri_bufferSize`, `hipsolverCpotri_bufferSize`, and
+      `hipsolverZpotri_bufferSize`
+  * `orgbr/ungbr`
+    * `hipsolverSorgbr`, `hipsolverDorgbr`, `hipsolverCungbr`, `hipsolverZungbr`
```
Comment (tfalders) [inline CHANGELOG.md]: Minor issue, but this line is missing the "and"

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,241 +1,328 @@
...
+    * `hipsolverSgeqrf`, `hipsolverDgeqrf`, `hipsolverCgeqrf`, and `hipsolverZgeqrf`
+    * `hipsolverSgeqrf_bufferSize`, `hipsolverDgeqrf_bufferSize`, `hipsolverCgeqrf_bufferSize`, and
+      `hipsolverZgeqrf_bufferSize`
+  * `gesvd`
+    * `hipsolverSgesvd`, `hipsolverDgesvd`, `hipsolverCgesvd`, and `hipsolverZgesvd`
+    * `hipsolverSgesvd_bufferSize`, `hipsolverDgesvd_bufferSize`, `hipsolverCgesvd_bufferSize`, and
+      `hipsolverZgesvd_bufferSize`
+  * `getrs`
+    * `hipsolverSgetrs`, `hipsolverDgetrs`, `hipsolverCgetrs`, and `hipsolverZgetrs`
+    * `hipsolverSgetrs_bufferSize, hipsolverDgetrs_bufferSize, hipsolverCgetrs_bufferSize, hipsolverZgetrs_bufferSize
```
Comment (tfalders) [inline CHANGELOG.md]: Missing quotation marks and the "and"

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,241 +1,328 @@
...
+* `hipsolverXgetrf` functions now take `lwork` as an argument
+
+### Removals
+
+* Removed unused `HIPSOLVER_FILL_MODE_FULL` enum value.
+* Removed `hipsolverComplex` and `hipsolverDoubleComplex` from the library; use `hipFloatComplex`
+  and `hipDoubleComplex` instead
 ### Removed
 - Removed unused HIPSOLVER_FILL_MODE_FULL enum value.
 - Removed hipsolverComplex and hipsolverDoubleComplex from the library. Use hipFloatComplex and hipDoubleComplex instead.
```
Comment (tfalders) [inline CHANGELOG.md]: These lines should be removed

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,241 +1,328 @@
...
 
 ## hipSOLVER 1.8.0 for ROCm 5.6.0
-### Added
-- Added compatibility API with hipsolverRf prefix
 
+### Additions
+
+* Added
+  [`hipsolverRf*`](https://rocm.docs.amd.com/projects/hipSOLVER/en/docs-5.6.1/userguide/usage.html#porting-cusolver-applications-to-hipsolver)
+  APIs for compatibility
```
Comment (tfalders) [inline CHANGELOG.md:78]: This is inconsistent with the compatibility API additions from 5.1 and 6.1. Please make them all consistent.

If the link is kept, wouldn't it make more sense to link to the API documentation? i.e., https://rocm.docs.amd.com/projects/hipSOLVER/en/latest/refactor-api/index.html

File: README.md [hunk 9]
```diff
@@ -1,14 +1,19 @@
...
-## Documentation
+hipSOLVER supports [rocSOLVER](https://github.com/ROCmSoftwarePlatform/rocSOLVER) and
+[cuSOLVER](https://developer.nvidia.com/cusolver) backends.
 
-For a detailed description of the hipSOLVER library, its implemented routines, the installation process and user guide, see the [hipSOLVER Documentation](https://rocm.docs.amd.com/projects/hipSOLVER/en/latest/index.html).
+## Documentation
 
-### How to build documentation
+Documentation for hipSOLVER is available at
+[https://rocm.docs.amd.com/projects/hipSOLVER/en/latest/](https://hipsolver.readthedocs.io/en/latest/).
```
Comment (tfalders) [inline README.md]: If I'm reading this correctly, the link is for rocm.docs.amd.com but it actually links to ReadTheDocs?

File: README.md [hunk 10]
```diff
@@ -18,24 +23,35 @@ pip3 install -r sphinx/requirements.txt
...
-To download the hipSOLVER source code, clone this repository with the command:
+To download the hipSOLVER source code, clone this repository using:
 
-    git clone https://github.com/ROCmSoftwarePlatform/hipSOLVER.git
+```bash
+git clone https://github.com/ROCmSoftwarePlatform/hipSOLVER.git
+```
 
-hipSOLVER requires either cuSOLVER or rocSOLVER + SuiteSparse to be installed on the system. Once these are installed, the following commands will build hipSOLVER and install to `/opt/rocm`:
+To install hipSOLVER, you must have cuSOLVER or rocSOLVER and rocBLAS installed on your system.
```
Comment (tfalders) [inline README.md]: SuiteSparse is also a required dependency as of 6.1. The full dependency tree for hipSOLVER is:
- rocSOLVER
  - rocBLAS
    - Tensile
  - rocSPARSE
    - rocPRIM
- SuiteSparse

**Review by tfalders — CHANGES_REQUESTED** [review]
As with https://github.com/ROCm/rocSOLVER/pull/635, I would like the different releases to be clearly demarcated in raw view, possibly by restoring the original newline spacing. I also have some other comments that should be addressed before we merge this PR.

File: CHANGELOG.md (hunk 1 above)
Comment (LisaDelaney) [reply CHANGELOG.md]: definitely! thanks for pointing this out

File: README.md (hunk 10 above)
Comment (LisaDelaney) [reply README.md]: made an update to this bit--mind reviewing it? I used the same formatting, but can restructure it to be more verbose if that would make more sense...

File: CHANGELOG.md (hunk 8 above)
Comment (LisaDelaney) [reply CHANGELOG.md:78]: If you prefer a complete list, can you provide it? (feel free to make commits/suggestions in this PR)

**Pull request closed without merging**